frontends/web/src/routes/account/send/services.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { alertUser } from '../../../components/alert/Alert';
+import { i18n } from '../../../i18n/i18n';
+
+export type TProposalError = {
+    addressError: string;
+    amountError: string;
+    feeError: string;
+}
+
+type TProposalErrorHandling = TProposalError | { proposedFee: undefined }
+
+export const txProposalErrorHandling = (registerEvents: () => void, unregisterEvents: () => void, errorCode?: string) => {
+  const { t } = i18n;
+  let errorHandling = {} as Partial<TProposalErrorHandling>;
+  switch (errorCode) {
+  case 'invalidAddress':
+    errorHandling = { addressError: t('send.error.invalidAddress') };
+    break;
+  case 'invalidAmount':
+  case 'insufficientFunds':
+    errorHandling = { amountError: t(`send.error.${errorCode}`), proposedFee: undefined };
+    break;
+  case 'feeTooLow':
+  case 'feesNotAvailable':
+    errorHandling = { feeError: t(`send.error.${errorCode}`) };
+    break;
```

Comment:
Looks like you could just `return` the object instead and drop errorHandling variable.

Reply:
Ah true, `proposedFee` isn't even an error anyway. So saving it in a variable called errorHandling isn't exactly accurate either. Ty!